app/src/main/java/com/example/dungeontest/NamingScreen.kt

```diff
@@ -101,31 +106,44 @@ fun NamingScreen(drawerState: DrawerState, scope: CoroutineScope, base64EncodedP
                 modifier = Modifier.fillMaxSize()
             ) {
                 Spacer(modifier = Modifier.height(2.dp))
-                TextInputField(mapInputNameVal,"Map Name")
+                TextInputField(mapInputNameVal, "Map Name")
                 Spacer(modifier = Modifier.weight(1f))
                 Button(
                     onClick = {
-                        Log.v("SaveButton", "button clicked")
+                        Log.v("NextButton", "button clicked")
                         scope.launch {
                             val mapName = mapInputNameVal.value.text
                             val exists = viewModel.mapExists(mapName)
                             if (exists) {
                                 showDialog.value = true
                             } else {
-                                // Store this value in case we want to check if it actually saved or not
-                                val didSaveMap = viewModel.insertMap(MapRecord(
-                                    mapName,
-                                    photoUri,
-                                ))
-                                navController.navigate("EditorScreen")
+                                Log.v("NamingScreen", photoPath)
+                                val file = File(photoPath)
+                                val imageBytes = file.readBytes()
+                                val imagePath = saveImageToInternalStorage(context = context, imageBytes = imageBytes, mapName = mapName)
+
+                                Log.v("NamingScreen", imagePath.toString())
+                                if (imagePath != null) {
+                                    val newMap = MapRecord(
+                                        mapName,
+                                        imagePath.toString(),
+                                        null
+                                    )
+                                    val didSaveMap = viewModel.insertMap(newMap)
```

Comment:
I'm confused. I thought we weren't actually going to save the map's information to the database until after they'd seen it in the editor and pressed the save button there?
Same question when it comes to the code for the Confirm button

Reply:
We haven't gotten to the actual saving part on the EditorScreen yet. This was just leftover from how it used to be